### PR TITLE
multiple workspaces management example

### PR DESCRIPTION
**Pull request by Uk1288**
## Description

This PR show an example of using the tfe provider to manage multiple workspaces with associated workspace variables. 

 _[Origin PR](https://github.com/hashicorp/terraform-provider-tfe/pull/129)_

## Testing plan

1.  Run the workspaces example config in the PR

## External links

- [Related PR](https://github.com/hashicorp/terraform-provider-tfe/pull/129)

### CODE REVIEW

**Review by laurenolivia — APPROVED** [review]
awesome ✨

File: examples/workspaces_variables/managed/main.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# Variables
+#
+
+variable "sens_tf_var" {
+  default = ""
+}
+
+variable "sens_env_var" {
+  default = ""
+}
```
Comment (sebasslash) [inline examples/workspaces_variables/managed/main.tf:14]: Are these supposed to be marked sensitive? Going of the `sens_` prefix.

File: examples/workspaces_variables/managed/main.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#
+# Output all of the variables for demonstration purposes
+#
+
+output "sens_tf_var" {
+  value = var.sens_tf_var
+}
```
Comment (sebasslash) [inline examples/workspaces_variables/managed/main.tf:46]: Same here

File: examples/workspaces_variables/managed/main.tf [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  default = "a default string"
+}
+
+# Value is initially supplied via some.auto.tfvars
+variable "a_number" {
+  type = number
+}
+
+variable "a_list" {
+  type = list
```
Comment (sebasslash) [inline examples/workspaces_variables/managed/main.tf:27]: Should be `list(<type>)`, same goes for `variable "a_map"`

File: examples/workspaces_variables/manager/main.tf [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+variable "vars_mapped_by_workspace_name" {
+    description = <<-EOT
+    This is the map of workspaces and variables. A workspace is created for each
+    top level key and then variables are set on the workspace.
+    EOT
+}
+
+variable "additional_vars" {
```
Comment (sebasslash) [inline examples/workspaces_variables/manager/main.tf:29]: These variables should specify their type. Given this is an example for those starting to manage TFC workspaces we should eliminate as many assumptions as we can.

**Review by sebasslash — CHANGES_REQUESTED** [review]
[Non-blocking feedback] There are tons of different ways to write a configuration -- personally I find this example to be a tad bit convoluted surrounding how variables are populated for the managed workspaces, especially if the assumption here is the customer is starting out on managing cloud workspaces. I'm not particularly fond of having one giant map being the source of truth to provision variables for each workspace -- but again, personal bias. Also it's possible to get away with using one github repository for both manager/managed configurations and simply changing the working directory. 

[Blocking feedback] Bias aside, there are some really small blocking items down below :arrow_down: 

That said, this is great work! :rocket:

**Comment by sebasslash** [comment]
Mentioning @brenthc for visibility

**Review by sebasslash — APPROVED** [review]

**Pull request merged**